Torch.Server/Views/Converters/ModToIdConverter.cs

```diff
@@ -32,9 +32,16 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
         {
             //if (targetType != typeof(int))
             //    throw new NotSupportedException("ModToIdConverter can only convert mods into int values or vise versa!");
-            var mod = (ModItemInfo) values[0];
-            var theModList = (MtObservableList<ModItemInfo>) values[1];
-            return theModList.IndexOf(mod);
+            if (values[0].GetType() != typeof(ModItemInfo) || values[1].GetType() != typeof(MtObservableList<ModItemInfo>))
```

Comment:
This is not how you check types in C#. Use the `is` keyword.

Reply:
I have modified it now to fix the checking of types